### PR TITLE
Fix inline style inheritance in Navigation block

### PR DESCRIPTION
**Pull request by tellthemachines**
## Description

Added the `style` attribute to context provided by the Navigation block, so that its child blocks can access custom colors set in the Navigation block. 

So far, with regards to the styles that can be set in Navigation block, this is only useful for `background-color`, as both `color` and all typography styles are inherited by child elements anyway. For this reason, I haven't explicitly added all the extra typography properties Navigation sets in addition to `font-size`.

By the same reasoning, we could probably remove the logic for setting inline `font-size` altogether. Thoughts?

Also, not sure if this will need a deprecation, as it seems this feature has been broken for a while, and previous changes to Navigation Link didn't add one.

## How has this been tested?

Add Navigation block and a few child items; create at least one submenu. Set a custom background color on Navigation and verify that submenu dropdown inherits that background color.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/navigation-link/index.php [hunk 1]
```diff
@@ -20,7 +20,7 @@ function block_core_navigation_link_build_css_colors( $context ) {
 
 	// Text color.
 	$has_named_text_color  = array_key_exists( 'textColor', $context );
-	$has_custom_text_color = array_key_exists( 'customTextColor', $context );
+	$has_custom_text_color = isset( $context['style']['color']['text'] );
```
Comment (talldan) [inline packages/block-library/src/navigation-link/index.php:23]: General comment that doesn't require any action in this PR—it'd be nice if there was a php function for handling the colors support setting.

Similarly, the amount of wrangling that has to be done of colors passed through context feels like something that could be improved.

File: packages/block-library/src/navigation-link/edit.js [hunk 2]
```diff
@@ -483,18 +462,17 @@ export default compose( [
 			hasSelectedInnerBlock,
 			getBlockParentsByBlockName,
 			getSelectedBlockClientId,
-			getSettings,
 		} = select( 'core/block-editor' );
 		const { clientId } = ownProps;
 		const rootBlock = head(
 			getBlockParentsByBlockName( clientId, 'core/navigation' )
 		);
 		const navigationBlockAttributes = getBlockAttributes( rootBlock );
```
Comment (talldan) [inline packages/block-library/src/navigation-link/edit.js:470]: Like the PR for Page List the other day, it should be possible to simplify this now to use the block `context` prop.

Can be left to a separate PR, but I think it'd be good to get to.

**Review by talldan — COMMENTED** [review]
This is working much better. I don't think a deprecation will be needed, as there's no change in attributes and the block is dynamic. As you say, it seems to have been broken for a while too.

The main issue I noticed when testing is that predefined colors don't apply to submenus when theme styles are disabled (not sure if that's in the scope of this PR, it's the same in `master`):
Screenshot 2021-02-11 at 9 38 40 am

Custom colors look good though 👍 
Screenshot 2021-02-11 at 9 37 31 am

----

As a separate observation, Twenty Twenty One seems to only style list items and doesn't handle the appender. Probably a theme issue more than anything.
Screenshot 2021-02-11 at 9 32 58 am

File: packages/block-library/src/navigation-link/edit.js (hunk 2 above)
Comment (tellthemachines) [reply packages/block-library/src/navigation-link/edit.js:470]: I'd rather do it as a separate PR together with converting the function to only using hooks, as we currently have an unseemly mix of hooks and HOCs in there 😅

**Comment by tellthemachines** [comment]
> The main issue I noticed when testing is that predefined colors don't apply to submenus when theme styles are disabled (not sure if that's in the scope of this PR, it's the same in master):

Turns out that was an easy fix (the styles are being applied inline to the Nav wrapper, so I only had to set its `ul`s and `li`s to inherit the background color).

> As a separate observation, Twenty Twenty One seems to only style list items and doesn't handle the appender. Probably a theme issue more than anything.

Yeah, this needs to be fixed in the theme. Cc @carolinan

**Review by talldan — APPROVED** [review]
Working perfectly now 🎉
